pb-rs/src/types.rs

```diff
@@ -1021,6 +1026,57 @@ impl Message {
         Ok(())
     }
 
+    fn write_impl_rentals<W: Write>(&self, w: &mut W) -> Result<()> {
+        write!(w,
+            r#"
+            rental! {{
+                mod mod_{name}Rental {{
+                    use super::*;
+
+                    #[rental(covariant, debug)]
+                    pub struct {name}Rental {{
+                        buf: Vec<u8>,
+                        proto: {name}<'buf>,
+                    }}
+                }}
+            }}
+            pub use self::mod_{name}Rental::{name}Rental;
+
+            impl TryFrom<Vec<u8>> for {name}Rental {{
+                type Error=quick_protobuf::Error;
+
+                fn try_from(buf: Vec<u8>) -> Result<Self> {{
+                    Self::try_new_or_drop(buf, |buf| {{
+                        let mut reader = BytesReader::from_bytes(&buf);
+                        //debug!("parsing {name}={{:?}}", buf);
+                        let proto = {name}::from_reader(&mut reader, &buf);
+                        //debug!("parsed {name}={{:?}}", proto);
+                        proto
+                    }})
+                }}
+            }}
+
+            #[cfg(feature = "test_helpers")]
```

Comment:
What do you think about having a generated code with a `#[cfg(feature = "test_helpers")]`? I've found it useful in tests to be able to easily create a rental. Thoughts on making this easy to use for others? I can add documentation and an example.